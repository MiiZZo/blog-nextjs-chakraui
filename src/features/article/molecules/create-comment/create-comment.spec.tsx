import React from "react";
import { myRender } from "@lib/testing/render";
import { cleanup, fireEvent } from "@testing-library/react";
import CreateComment from "./create-comment";

afterEach(cleanup);

describe("create comment component", () => {
  const handleDiscardComment = jest.fn();
  const handleSubmitComment = jest.fn();

  it("should call the handleDiscardComment function when modal open", () => {
    const handleDiscardComment = jest.fn();
    const { getByTestId } = myRender(
      <CreateComment
        handleDiscardComment={handleDiscardComment}
        handleSubmitComment={handleSubmitComment}
      />
    );
    const closeButton = getByTestId("close");
    const input = getByTestId("textarea");

    fireEvent.change(input, { target: { value: "my value" } });
    fireEvent.click(closeButton);

    const discardButton = getByTestId("discard");

    fireEvent.click(discardButton);

    expect(handleDiscardComment).toBeCalled();
  });

  it("should call the handleSubmitComment function", () => {
    const handleSubmitComment = jest.fn();
    const { getByTestId } = myRender(
      <CreateComment
        handleSubmitComment={handleSubmitComment}
        handleDiscardComment={handleDiscardComment}
      />
    );
    const submitButton = getByTestId("submit");

    fireEvent.click(submitButton);

    expect(handleSubmitComment).toBeCalledTimes(1);
  });

  it("should call the handleDiscardComment function", () => {
    const handleDiscardComment = jest.fn();
    const { getByTestId } = myRender(
      <CreateComment
        handleSubmitComment={handleSubmitComment}
        handleDiscardComment={handleDiscardComment}
      />
    );
    const closeButton = getByTestId("close");

    fireEvent.click(closeButton);

    expect(handleDiscardComment).toBeCalledTimes(1);
  });

  it("should open modal", () => {
    const handleDiscardComment = jest.fn();
    const value = "my value";
    const { asFragment, getByTestId } = myRender(
      <CreateComment
        handleSubmitComment={handleSubmitComment}
        handleDiscardComment={handleDiscardComment}
      />
    );
    const input = getByTestId("textarea");
    const discardButton = getByTestId("close");

    fireEvent.change(input, { target: { value } });
    fireEvent.click(discardButton);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should match a shapshot", () => {
    const { asFragment } = myRender(
      <CreateComment
        handleSubmitComment={handleSubmitComment}
        handleDiscardComment={handleDiscardComment}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
