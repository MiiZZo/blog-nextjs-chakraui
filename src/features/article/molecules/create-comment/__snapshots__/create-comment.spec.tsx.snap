// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create comment component should match a shapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1cl7zdz"
  >
    <div
      class="css-kjafn5"
    >
      <textarea
        class="css-jq13t9"
        data-testid="textarea"
        name="comment"
        placeholder="Write a comment..."
      />
      <div
        class="css-1f4xxbf"
      >
        <div
          aria-valuemax="500"
          aria-valuemin="0"
          aria-valuenow="0"
          class="css-1bfqfh5"
          data-testid="circular-indicator"
          role="progressbar"
        >
          <svg
            class="css-24ggqe"
            viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
          >
            <circle
              class="css-9y0bww"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-track="true"
              r="50"
              stroke-dasharray="314.159"
              stroke-dashoffset="0"
              stroke-width="11.111111111111112"
            />
            <circle
              class="css-ich4kh"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-indicator="true"
              r="50"
              stroke-dasharray="314.159"
              stroke-dashoffset="314.1592653589793"
              stroke-width="11.111111111111112"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      class="css-skyqpb"
      data-testid="submit"
      type="button"
    >
      Submit comment
    </button>
    <button
      class="css-w8jeta"
      data-testid="close"
      type="button"
    >
      Close
    </button>
  </div>
</DocumentFragment>
`;

exports[`create comment component should open modal 1`] = `
<DocumentFragment>
  <div
    class="css-1cl7zdz"
  >
    <div
      class="css-kjafn5"
    >
      <textarea
        class="css-jq13t9"
        data-testid="textarea"
        name="comment"
        placeholder="Write a comment..."
      >
        my value
      </textarea>
      <div
        class="css-1f4xxbf"
      >
        <div
          aria-valuemax="500"
          aria-valuemin="0"
          aria-valuenow="8"
          class="css-1bfqfh5"
          data-testid="circular-indicator"
          role="progressbar"
        >
          <svg
            class="css-24ggqe"
            viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
          >
            <circle
              class="css-9y0bww"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-track="true"
              r="50"
              stroke-dasharray="314.159"
              stroke-dashoffset="0"
              stroke-width="11.111111111111112"
            />
            <circle
              class="css-ich4kh"
              cx="111.11111111111111"
              cy="111.11111111111111"
              data-progress-indicator="true"
              r="50"
              stroke-dasharray="314.159"
              stroke-dashoffset="309.1327171132356"
              stroke-width="11.111111111111112"
            />
          </svg>
        </div>
      </div>
    </div>
    <button
      class="css-skyqpb"
      data-testid="submit"
      type="button"
    >
      Submit comment
    </button>
    <button
      class="css-w8jeta"
      data-testid="close"
      type="button"
    >
      Close
    </button>
  </div>
</DocumentFragment>
`;
